packages/flutter/lib/src/material/expansion_tile.dart

```diff
@@ -668,13 +668,51 @@ class _ExpansionTileState extends State<ExpansionTile> with SingleTickerProvider
     );
   }
 
+  @override
+  void didUpdateWidget(covariant ExpansionTile oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    final ThemeData theme = Theme.of(context);
+    final ExpansionTileThemeData expansionTileTheme = ExpansionTileTheme.of(context);
+    final ExpansionTileThemeData defaults = theme.useMaterial3
+      ? _ExpansionTileDefaultsM3(context)
+      : _ExpansionTileDefaultsM2(context);
+    if (widget.collapsedShape != oldWidget.collapsedShape
+      || widget.shape != oldWidget.shape) {
+      _updateShapeBorder(theme, expansionTileTheme, defaults);
+    }
+    if (widget.collapsedTextColor != oldWidget.collapsedTextColor
+      || widget.textColor != oldWidget.textColor) {
+      _updateHeaderColor(expansionTileTheme, defaults);
+    }
+    if (widget.collapsedIconColor != oldWidget.collapsedIconColor
+      || widget.iconColor != oldWidget.iconColor) {
+      _updateIconColor(expansionTileTheme, defaults);
+    }
+    if (widget.backgroundColor != oldWidget.backgroundColor
+      || widget.collapsedBackgroundColor != oldWidget.collapsedBackgroundColor) {
+      _updateBackgroundColor(expansionTileTheme);
+    }
+  }
+
   @override
   void didChangeDependencies() {
     final ThemeData theme = Theme.of(context);
     final ExpansionTileThemeData expansionTileTheme = ExpansionTileTheme.of(context);
     final ExpansionTileThemeData defaults = theme.useMaterial3
       ? _ExpansionTileDefaultsM3(context)
       : _ExpansionTileDefaultsM2(context);
+    _updateShapeBorder(theme, expansionTileTheme, defaults);
+    _updateHeaderColor(expansionTileTheme, defaults);
+    _updateIconColor(expansionTileTheme, defaults);
+    _updateBackgroundColor(expansionTileTheme);
+    super.didChangeDependencies();
+  }
+
+  void _updateShapeBorder(
+    ThemeData theme,
+    ExpansionTileThemeData expansionTileTheme,
+    ExpansionTileThemeData defaults,
```

Comment:
This parameter isn't used, but presumably it should be?

Reply:
Great catch, it should only `expansionTileTheme` and `themeData` fallback


https://github.com/flutter/flutter/blob/e308722e08b7552776aba7cc9b12a21a04a6bf9e/packages/flutter/lib/src/material/expansion_tile.dart#L686-L689